### PR TITLE
stm32: entropy: Enhance documentation around domain clock configuration issues

### PR DESCRIPTION
**Pull request by erwango**
Enhance mechanism around domain clock configuration issues
Add some more documentation

Should answer issues such as #56494

### CODE REVIEW

File: dts/bindings/rng/st,stm32-rng.yaml [hunk 1]
```diff
@@ -13,7 +13,13 @@ properties:
...
   clocks:
     required: true
-
+    description: |
+         Specifies the clock line to be enabled in clock controller as well as
+         the clock domain used, for instance:
+                   <&rcc STM32_SRC_MSI CLK48_SEL(3)> /* RNG clock domain set to MSI */
+         A correctly configured domain clock is required to allow the integrated low
+         sampling clock detection mecanism to behave properly.
+         In provided example, MSI should be configured to provide 48Mhz clock.
```
Comment (gautierg-st) [inline dts/bindings/rng/st,stm32-rng.yaml:22]: Nitpick: Add a blank line. Also mecanism -> mechanism

**Review by gautierg-st — CHANGES_REQUESTED** [review]
Some nitpicks, otherwise LGTM

File: dts/bindings/rng/st,stm32-rng.yaml (hunk 1 above)
Comment (erwango) [reply dts/bindings/rng/st,stm32-rng.yaml:22]: Fixed, thanks

**Review by gautierg-st — APPROVED** [review]

**Review by GeorgeCGV — APPROVED** [review]
LGTM, except for a minor typo.

File: dts/bindings/rng/st,stm32-rng.yaml [hunk 2]
```diff
@@ -13,6 +13,13 @@ properties:
 
   clocks:
     required: true
+    description: |
+         Specifies the clock line to be enabled in clock controller as well as
+         the clock domain used, for instance:
+                   <&rcc STM32_SRC_MSI CLK48_SEL(3)> /* RNG clock domain set to MSI */
+         A correctly configured domain clock is required to allow the integrated low
+         sampling clock detection mecanism to behave properly.
```
[the remark this answers is not in the document]
Comment (erwango) [reply dts/bindings/rng/st,stm32-rng.yaml:21]: Seeing this too late. Thanks, will fix when possible

**Pull request merged**